config/provider.go

```diff
@@ -20,6 +20,13 @@
 
 package config
 
+import (
+	"context"
+)
+
 type Provider interface {
-	Load(cfg Config) error
+	Register(ctx context.Context, service string, cfg interface{})
+	Get(ctx context.Context, service string) (interface{}, error)
```

Comment:
Given we now have go 1.18, I wonder if this interface{} could be replaced with a `<T>`

Reply:
Yup. Replaced with `any`.